### ISSUE TITLE
Error "Only RSA or EC key is supported." on issue of new cert.

### ISSUE THREAD

**Issue by jonl-percsolutions-com** [acmesh-official/acme.sh#3659]
Attempting to issue a new certificate on a new domain name using godaddy dns.  Renewal is failing with an error "Only RSA or EC key is supported. "  I have previously issued certs on this server, last cert renewal was successful just a few days ago.  I saw previous issues with similar errors, but none of them seem to be the same.  The account.key is not empty, deleting the account key does not fix the issue.  Forcing between the different directories (zerossl or letsencrypt) does not change the issue.

Tried reverting all the way back to 2.8.8 with no success.  Any assistance would be appreciated.

Steps to reproduce
------------------

Issue a new cert using go daddy dns.

Debug log
-----------------

[root@localhost ~]# acme.sh --issue --dns dns_gd -d my.domain.com --debug 2
[Wed Aug 11 16:15:10 EDT 2021] Lets find script dir.
[Wed Aug 11 16:15:10 EDT 2021] _SCRIPT_='/root/.acme.sh/acme.sh'
[Wed Aug 11 16:15:10 EDT 2021] _script='/root/.acme.sh/acme.sh'
[Wed Aug 11 16:15:10 EDT 2021] _script_home='/root/.acme.sh'
[Wed Aug 11 16:15:10 EDT 2021] Using config home:/root/.acme.sh
[Wed Aug 11 16:15:10 EDT 2021] LE_WORKING_DIR='/root/.acme.sh'
https://github.com/acmesh-official/acme.sh
v3.0.1
[Wed Aug 11 16:15:10 EDT 2021] Running cmd: issue
[Wed Aug 11 16:15:10 EDT 2021] _main_domain='my.domain.com'
[Wed Aug 11 16:15:10 EDT 2021] _alt_domains='no'
[Wed Aug 11 16:15:10 EDT 2021] Using config home:/root/.acme.sh
[Wed Aug 11 16:15:10 EDT 2021] default_acme_server
[Wed Aug 11 16:15:10 EDT 2021] ACME_DIRECTORY='https://acme.zerossl.com/v2/DV90'
[Wed Aug 11 16:15:10 EDT 2021] _ACME_SERVER_HOST='acme.zerossl.com'
[Wed Aug 11 16:15:10 EDT 2021] _ACME_SERVER_PATH='v2/DV90'
[Wed Aug 11 16:15:10 EDT 2021] DOMAIN_PATH='/root/.acme.sh/my.domain.com'
[Wed Aug 11 16:15:10 EDT 2021] 'dns_gd' does not contain 'dns'
[Wed Aug 11 16:15:10 EDT 2021] Using ACME_DIRECTORY: https://acme.zerossl.com/v2/DV90
[Wed Aug 11 16:15:10 EDT 2021] _init api for server: https://acme.zerossl.com/v2/DV90
[Wed Aug 11 16:15:10 EDT 2021] Retrying GET
[Wed Aug 11 16:15:10 EDT 2021] GET
[Wed Aug 11 16:15:10 EDT 2021] url='https://acme.zerossl.com/v2/DV90'
[Wed Aug 11 16:15:10 EDT 2021] timeout=
[Wed Aug 11 16:15:10 EDT 2021] displayError='1'
[Wed Aug 11 16:15:10 EDT 2021] _CURL='curl --silent --dump-header /root/.acme.sh/http.header  -L  --trace-ascii /tmp/tmp.VuhvYoEDFm  -g '
[Wed Aug 11 16:15:10 EDT 2021] ret='0'
[Wed Aug 11 16:15:10 EDT 2021] _hcode='0'
[Wed Aug 11 16:15:10 EDT 2021] response='{
  "newNonce": "https://acme.zerossl.com/v2/DV90/newNonce",
  "newAccount": "https://acme.zerossl.com/v2/DV90/newAccount",
  "newOrder": "https://acme.zerossl.com/v2/DV90/newOrder",
  "revokeCert": "https://acme.zerossl.com/v2/DV90/revokeCert",
  "keyChange": "https://acme.zerossl.com/v2/DV90/keyChange",
  "meta": {
    "termsOfService": "https://secure.trust-provider.com/repository/docs/Legacy/20201020_Certificate_Subscriber_Agreement_v_2_4_click.pdf",
    "website": "https://zerossl.com",
    "caaIdentities": ["sectigo.com", "trust-provider.com", "usertrust.com", "comodoca.com", "comodo.com"],
    "externalAccountRequired": true
  }
}'
[Wed Aug 11 16:15:10 EDT 2021] ACME_KEY_CHANGE='https://acme.zerossl.com/v2/DV90/keyChange'
[Wed Aug 11 16:15:10 EDT 2021] ACME_NEW_AUTHZ
[Wed Aug 11 16:15:10 EDT 2021] ACME_NEW_ORDER='https://acme.zerossl.com/v2/DV90/newOrder'
[Wed Aug 11 16:15:10 EDT 2021] ACME_NEW_ACCOUNT='https://acme.zerossl.com/v2/DV90/newAccount'
[Wed Aug 11 16:15:10 EDT 2021] ACME_REVOKE_CERT='https://acme.zerossl.com/v2/DV90/revokeCert'
[Wed Aug 11 16:15:10 EDT 2021] ACME_AGREEMENT='https://secure.trust-provider.com/repository/docs/Legacy/20201020_Certificate_Subscriber_Agreement_v_2_4_click.pdf'
[Wed Aug 11 16:15:10 EDT 2021] ACME_NEW_NONCE='https://acme.zerossl.com/v2/DV90/newNonce'
[Wed Aug 11 16:15:11 EDT 2021] Le_NextRenewTime
[Wed Aug 11 16:15:11 EDT 2021] Using CA: https://acme.zerossl.com/v2/DV90
[Wed Aug 11 16:15:11 EDT 2021] _on_before_issue
[Wed Aug 11 16:15:11 EDT 2021] _chk_main_domain='my.domain.com'
[Wed Aug 11 16:15:11 EDT 2021] _chk_alt_domains
[Wed Aug 11 16:15:11 EDT 2021] 'dns_gd' does not contain 'no'
[Wed Aug 11 16:15:11 EDT 2021] Le_LocalAddress
[Wed Aug 11 16:15:11 EDT 2021] d='my.domain.com'
[Wed Aug 11 16:15:11 EDT 2021] Check for domain='my.domain.com'
[Wed Aug 11 16:15:11 EDT 2021] _currentRoot='dns_gd'
[Wed Aug 11 16:15:11 EDT 2021] d
[Wed Aug 11 16:15:11 EDT 2021] 'dns_gd' does not contain 'apache'
[Wed Aug 11 16:15:11 EDT 2021] config file is empty, can not read CA_KEY_HASH
[Wed Aug 11 16:15:11 EDT 2021] _saved_account_key_hash
[Wed Aug 11 16:15:11 EDT 2021] Using config home:/root/.acme.sh
[Wed Aug 11 16:15:11 EDT 2021] ACME_DIRECTORY='https://acme.zerossl.com/v2/DV90'
[Wed Aug 11 16:15:11 EDT 2021] _ACME_SERVER_HOST='acme.zerossl.com'
[Wed Aug 11 16:15:11 EDT 2021] _ACME_SERVER_PATH='v2/DV90'
[Wed Aug 11 16:15:11 EDT 2021] _init api for server: https://acme.zerossl.com/v2/DV90
[Wed Aug 11 16:15:11 EDT 2021] keyfile=/root/.acme.sh/ca/acme.zerossl.com/v2/DV90/account.key
[Wed Aug 11 16:15:11 EDT 2021] -----BEGIN PRIVATE KEY-----
..deleted...
-----END PRIVATE KEY-----
[Wed Aug 11 16:15:11 EDT 2021] _on_issue_err
[Wed Aug 11 16:15:11 EDT 2021] Please check log file for more details: /root/.acme.sh/acme.sh.log
[Wed Aug 11 16:15:11 EDT 2021] _chk_vlist
[Wed Aug 11 16:15:11 EDT 2021] socat doesn't exist.
[Wed Aug 11 16:15:11 EDT 2021] Diagnosis versions:
openssl:openssl
OpenSSL 1.0.2k-fips  26 Jan 2017
apache:
apache doesn't exist.
nginx:
nginx version: nginx/1.20.1
built by gcc 4.8.5 20150623 (Red Hat 4.8.5-44) (GCC)
built with OpenSSL 1.1.1g FIPS  21 Apr 2020
TLS SNI support enabled
configure arguments: --prefix=/usr/share/nginx --sbin-path=/usr/sbin/nginx --modules-path=/usr/lib64/nginx/modules --conf-path=/etc/nginx/nginx.conf --error-log-path=/var/log/nginx/error.log --http-log-path=/var/log/nginx/access.log --http-client-body-temp-path=/var/lib/nginx/tmp/client_body --http-proxy-temp-path=/var/lib/nginx/tmp/proxy --http-fastcgi-temp-path=/var/lib/nginx/tmp/fastcgi --http-uwsgi-temp-path=/var/lib/nginx/tmp/uwsgi --http-scgi-temp-path=/var/lib/nginx/tmp/scgi --pid-path=/run/nginx.pid --lock-path=/run/lock/subsys/nginx --user=nginx --group=nginx --with-compat --with-debug --with-file-aio --with-google_perftools_module --with-http_addition_module --with-http_auth_request_module --with-http_dav_module --with-http_degradation_module --with-http_flv_module --with-http_gunzip_module --with-http_gzip_static_module --with-http_image_filter_module=dynamic --with-http_mp4_module --with-http_perl_module=dynamic --with-http_random_index_module --with-http_realip_module --with-http_secure_link_module --with-http_slice_module --with-http_ssl_module --with-http_stub_status_module --with-http_sub_module --with-http_v2_module --with-http_xslt_module=dynamic --with-mail=dynamic --with-mail_ssl_module --with-pcre --with-pcre-jit --with-stream=dynamic --with-stream_ssl_module --with-stream_ssl_preread_module --with-threads --with-cc-opt='-O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic' --with-ld-opt='-Wl,-z,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -Wl,-E'

**Comment by jonl-percsolutions-com** [comment]
from log file:
[Wed Aug 11 16:15:11 EDT 2021] _chk_alt_domains
[Wed Aug 11 16:15:11 EDT 2021] Le_LocalAddress
[Wed Aug 11 16:15:11 EDT 2021] d='my.domain.com'
[Wed Aug 11 16:15:11 EDT 2021] Check for domain='my.domain.com'
[Wed Aug 11 16:15:11 EDT 2021] _currentRoot='dns_gd'
[Wed Aug 11 16:15:11 EDT 2021] d
[Wed Aug 11 16:15:11 EDT 2021] config file is empty, can not read CA_KEY_HASH
[Wed Aug 11 16:15:11 EDT 2021] Using config home:/root/.acme.sh
[Wed Aug 11 16:15:11 EDT 2021] ACME_DIRECTORY='https://acme.zerossl.com/v2/DV90'
[Wed Aug 11 16:15:11 EDT 2021] _init api for server: https://acme.zerossl.com/v2/DV90
[Wed Aug 11 16:15:11 EDT 2021] Only RSA or EC key is supported. keyfile=/root/.acme.sh/ca/acme.zerossl.com/v2/DV90/account.key
[Wed Aug 11 16:15:11 EDT 2021] _on_issue_err
[Wed Aug 11 16:15:11 EDT 2021] Please check log file for more details: /root/.acme.sh/acme.sh.log
[Wed Aug 11 16:15:11 EDT 2021] socat doesn't exist.
[Wed Aug 11 16:15:11 EDT 2021] Diagnosis versions:
openssl:openssl
OpenSSL 1.0.2k-fips  26 Jan 2017
apache:
apache doesn't exist.
nginx:
nginx version: nginx/1.20.1

**Comment by SteveGilvarry** [comment]
I was seeing the same thing, tried zerossl and letsencrypt just like you.

Traced the error message to https://github.com/acmesh-official/acme.sh/blob/d96cca382274c7595785a94816ef586e452b7f27/acme.sh#L1707
The account.key file for each CA was coming back as BEGIN PRIVATE KEY, so did not match either of the checks for what type it was. Found [elsewhere](https://github.com/acmesh-official/acme.sh/blob/6bdf689d0f14e27fe41a3064d98a57d9a61c7565/acme.sh#L1043) that the RSA check had been patched to accept BEGIN PRIVATE KEY, so gave it a go. No idea if this is a legit fix, but it got me working as far as I can tell.
Change I made was
https://github.com/acmesh-official/acme.sh/blob/d96cca382274c7595785a94816ef586e452b7f27/acme.sh#L1616
`  if grep "BEGIN RSA PRIVATE KEY" "$keyfile" >/dev/null 2>&1 || grep "BEGIN PRIVATE KEY" "$keyfile" >/dev/null 2>&1; then`

Issue alluded to here helped me https://github.com/acmesh-official/acme.sh/issues/3399#issuecomment-785136753

**Comment by bwyx** [comment]
> I was seeing the same thing, tried zerossl and letsencrypt just like you.

it seems only zerossl that causing the error. I used letsencrypt server and everything just work, did you type the command properly? @SteveGilvarry 


@jonl-percsolutions-com try to use different server
_https://github.com/acmesh-official/acme.sh/wiki/Server_

**Comment by jonl-percsolutions-com** [comment]
@bwyx the only thing that appears to work, is to add in the OR operation on line 1618 to the acme.sh script as indicated by @SteveGilvarry in https://github.com/acmesh-official/acme.sh/issues/3659#issuecomment-898993215

**Comment by mikatuo** [comment]
Perhaps it makes sense to add a note to the error message to try using a different CA?

TL; DR;

In my case using Let's Encrypt CA helped, but it took quite some time to figure that out. I thought that my VPS is the reason (I even tried Certbot and also got an error message so I was 100% sure that it's the VPS).

Had a similar problem when running:

```bash
acme.sh --issue -d my.domain.com -w /var/www/html
[Fri 07 Jun 2024 02:35:33 AM CDT] Using CA: https://acme.zerossl.com/v2/DV90
[Fri 07 Jun 2024 02:35:33 AM CDT] Only RSA or EC key is supported. keyfile=/root/.acme.sh/ca/acme.zerossl.com/v2/DV90/account.key
[Fri 07 Jun 2024 02:35:33 AM CDT] Please check log file for more details: /root/.acme.sh/acme.sh.log

```

Then used Let's Encrypt server and it worked:

```bash
acme.sh --issue -d my.domain.com -w /var/www/html --server letsencrypt
```
